src/SignalR/common/Protocols.MessagePack/src/Protocol/MessagePackHubProtocol.cs

```diff
@@ -54,33 +54,40 @@ public MessagePackHubProtocol(IOptions<MessagePackHubProtocolOptions> options)
         {
             var msgPackOptions = options.Value;
             SetupResolver(msgPackOptions);
-            _msgPackSerializerOptions.WithSecurity(MessagePackSecurity.UntrustedData);
-        }
 
-        private void SetupResolver(MessagePackHubProtocolOptions options)
-        {
-            // if counts don't match then we know users customized resolvers so we set up the options
-            // with the provided resolvers
-            if (options.FormatterResolvers.Count != SignalRResolver.Resolvers.Count)
+            void SetupResolver(MessagePackHubProtocolOptions options)
```

Comment:
What's the value of having this in a local function now? Why not just inline it?

Reply:
you mean putting all the code in the `ctor` ?
I can totally do that, i wondered that if it existed before, it could have been like this for readability

Do i remove the local function then ?

Reply:
and my point was about `hasCustomFormatterResovler`
as the previous code was basically doing the same thing 2, 3 times

Reply:
I also could have done 
```
IFormatterResolver resolve = null;

/* LOGIC HERE */
resolver = hasCustomFormatterResovler
                  ? resolver = CompositeResolver.Create(Array.Empty<IMessagePackFormatter>(), (IReadOnlyList<IFormatterResolver>)msgPackOptions.FormatterResolvers);
                  : SignalRResolver.Instance;
```

but it introduced a `null` around, not sure what's the best over all